test/functional/apps/discover/_doc_table.ts

```diff
@@ -0,0 +1,156 @@
+  const PageObjects = getPageObjects(['common', 'discover', 'header', 'timePicker']);
+  const defaultSettings = {
+    defaultIndex: 'logstash-*',
+  };
+
+  describe('discover doc table', function describeIndexTests() {
+    const defaultRowsLimit = 50;
+    const rowsHardLimit = 500;
+
+    before(async function () {
+      log.debug('load kibana index with default index pattern');
+      await esArchiver.load('discover');
+
+      // and load a set of makelogs data
+      await esArchiver.loadIfNeeded('logstash_functional');
+      await kibanaServer.uiSettings.replace(defaultSettings);
+      log.debug('discover doc table');
+      await PageObjects.common.navigateToApp('discover');
+    });
+
+    beforeEach(async function () {
+      await PageObjects.timePicker.setDefaultAbsoluteRange();
+    });
+
+    it('should show the first 50 rows by default', async function () {
+      // with the default range the number of hits is ~14000
+      const rows = await PageObjects.discover.getDocTableRows();
+      expect(rows.length).to.be(defaultRowsLimit);
+    });
+
+    it('should refresh the table content when changing time window', async function () {
+      const initialRows = await PageObjects.discover.getDocTableRows();
+
+      const fromTime = 'Sep 20, 2015 @ 23:00:00.000';
+      const toTime = 'Sep 20, 2015 @ 23:14:00.000';
+
+      await PageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+      await PageObjects.discover.waitUntilSearchingHasFinished();
+
+      const finalRows = await PageObjects.discover.getDocTableRows();
+      expect(finalRows.length).to.be.below(initialRows.length);
+    });
+
+    it(`should load up to ${rowsHardLimit} rows whin scrolling at the end of the table`, async function () {
+      const initialRows = await PageObjects.discover.getDocTableRows();
+      // click the Skip to the end of the table
+      await PageObjects.discover.skipToEndOfDocTable();
+      // now count the rows
+      const finalRows = await PageObjects.discover.getDocTableRows();
+      expect(finalRows.length).to.be.above(initialRows.length);
+      expect(finalRows.length).to.be(rowsHardLimit);
+    });
+
+    it('should go the end of the table when using the accessible Skip button', async function () {
+      // click the Skip to the end of the table
+      await PageObjects.discover.skipToEndOfDocTable();
+      // now check the footer text content
+      const footer = await PageObjects.discover.getDocTableFooter();
+      log.debug(await footer.getVisibleText());
+      expect(await footer.getVisibleText()).to.have.string(rowsHardLimit);
+    });
+
+    describe('expand a document row', function () {
+      beforeEach(async function () {
+        await PageObjects.discover.expandToggleDocTableRow(1);
+      });
+
+      it('should expand the detail row when the toggle arrow is clicked', async function () {
+        await retry.try(async function () {
+          await PageObjects.discover.expandToggleDocTableRow(1);
```

Comment:
I might be wrong , but isn't it the case here, that before you run the test, you're expanding a document , which means, that the first run it will fail, because the document is already expanded, and a 2nd execution will undo the expansion? if this is the case, it might make send to improve `expandToggleDocTableRow` it expand the doc only when needed (when it's not expanded)?

Reply:
True.
In the test it wasn't happening because when I was trying to get the content of it, a check was in place to verify the state of the toggle, and a new click was performed in case it was closed.
I'll refactor it to correctly handle the state in this area.